Content.Shared/Singularity/Components/SharedEventHorizonComponent.cs

```diff
@@ -0,0 +1,51 @@
+using Content.Shared.Singularity.EntitySystems;
+
+namespace Content.Shared.Singularity.Components;
+
+public abstract class SharedEventHorizonComponent : Component
+{
+    /// <summary>
+    ///     The radius of the event horizon within which it will destroy all entities and tiles.
+    ///     If < 0.0 this behavior will not be active.
+    /// </summary>
+    [Access(friends:typeof(SharedEventHorizonSystem))]
+    public float _radius;
+
+    /// <summary>
+    ///     Whether the event horizon can consume/destroy the devices built to contain it.
+    /// </summary>
+    [Access(friends:typeof(SharedEventHorizonSystem))]
+    public bool _canBreachContainment = false;
+
+    /// <summary>
+    ///     The public getter/setter for _radius.
+    ///     Delegates setting to SharedEventHorizonSystem.
+    /// </summary>
+    [DataField("radius")]
+    [ViewVariables(VVAccess.ReadWrite)]
+    public float Radius
+    {
+        get => _radius;
+        set { IoCManager.Resolve<IEntitySystemManager>().GetEntitySystem<SharedEventHorizonSystem>().SetRadius(Owner, value, eventHorizon: this); }
```

Comment:
absolutely not. use a VV fake setter if thats what this is for, if this is for other things in code to access then use access and make people go through the system method

Reply:
I'm not certain what a VV fake setter is.

Reply:
I've moved the VVAccess bits to a new property that's just for VV.

Reply:
I have been informed what VV fake setters are. Will do.

Reply:
Done